Merge CDT Cloud update to 1.49.1

ndoschek opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md
-->

#### What it does
- Updates CDT cloud Blueprint to Theia Blueprint 1.49.101 (bfa92d)
- Updates theia-traceviewer to 0.2.4
- Fix a few non-critical Docker warnings in the Dockerfile
- Update to Node 20
- Copy CDTCloudBlueprintLogo.svg and adapt to splashScreen resolution and use as spashScreen image (CDTCloudBlueprintSplash.svg) for electron applications
- Copy applications/electron/resources/icons/WindowsLauncherIcon/windows ICO0-24bit/48-48.ico and use as applications/browser/ico/favicon.ico

Part of https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/issues/89

Contributed on behalf of STMicroelectronics

Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>

#### How to test
<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->

- Test electron application (packaged), browser application and docker application.
- Also run E2E tests (`yarn electron package:preview && yarn electron test`)

ndoschek commented:
Hi @sdirix , could you please have a look at the update PR once you're back from vacation, TIA

sdirix commented on Jenkinsfile:
```diff
@@ -261,7 +261,7 @@ spec:
                             script {
                                 uploadInstaller('windows')
-                                copyInstallerAndUpdateLatestYml('windows', 'CDTCloudBlueprintSetup', 'exe', 'latest.yml', '1.40.1')
+                                copyInstallerAndUpdateLatestYml('windows', 'CDTCloudBlueprintSetup', 'exe', 'latest.yml', '1.46.0,1.46.100,1.47.0,1.47.100,1.48.0,1.48.300,1.49.100')
```
I think we should only list versions here which we actually released, see [here](https://download.eclipse.org/theia/cdt-cloud/)

ndoschek replied:
Thanks for the link, I've updated the list of released versions accordingly.

sdirix commented on PUBLISHING.md:
We should delete the whole file as it doesn't fit to CDT Cloud Blueprint

sdirix commented on theia-extensions/updater/src/electron-main/update/theia-updater-impl.ts:
```diff
@@ -14,6 +14,15 @@ import * as path from 'path';
 import { injectable } from '@theia/core/shared/inversify';
+import { isOSX, isWindows } from '@theia/core';
+
+const STABLE_CHANNEL_WINDOWS = 'https://download.eclipse.org/theia/ide/version/windows';
+const STABLE_CHANNEL_MACOS = 'https://download.eclipse.org/theia/ide/latest/macos';
+const STABLE_CHANNEL_LINUX = 'https://download.eclipse.org/theia/ide/latest/linux';
+
+const PREVIEW_CHANNEL_WINDOWS = 'https://download.eclipse.org/theia/ide-preview/version/windows';
+const PREVIEW_CHANNEL_MACOS = 'https://download.eclipse.org/theia/ide-preview/latest/macos';
+const PREVIEW_CHANNEL_LINUX = 'https://download.eclipse.org/theia/ide-preview/latest/linux';
```
This does not align with CDT Cloud Blueprint. I think we don't want to merge in these changes but keep the file as is

sdirix commented on theia-extensions/updater/src/electron-browser/updater/theia-updater-preferences.ts:
```diff
@@ -16,6 +16,11 @@ export const theiaUpdaterPreferenceSchema: PreferenceSchema = {
             default: true
-        }
+        },
+        'updates.channel': {
+            type: 'string',
+            enum: ['stable', 'preview'], // once we have a nightly/next build, we can add a third channel
+            default: 'stable'
+        },
```
there is no stable/preview for cdt cloud blueprint

sdirix commented on theia-extensions/updater/src/electron-browser/updater/theia-updater-frontend-contribution.ts:
```diff
@@ -136,6 +136,13 @@ export class TheiaUpdaterFrontendContribution implements CommandContribution, Me
 
     @postConstruct()
     protected init(): void {
+        this.preferenceService.ready.then(() => this.updater.setUpdateChannel(this.preferenceService.get('updates.channel', 'stable')));
+        this.preferenceService.onPreferenceChanged(e => {
+            if (e.preferenceName === 'updates.channel') {
+                this.updater.setUpdateChannel(this.preferenceService.get('updates.channel', 'stable'));
+            }
+        });
+
```
same here

sdirix commented on releng/promote/Jenkinsfile:
I think we don't need this file

sdirix reviewed (changes requested):
Thanks for the work. Locally it works however I think it will break the Jenkinsbuild as well as the update mechanism. Please have a look.

ndoschek commented:
Thank you for the review @sdirix! I'll pushed an update addressing your comments.
Also, i quickly adapted the name of the splash svg image to match the CDTCloudBlueprint wording.

sdirix reviewed (approved):
LGTM and works for me
